pkg/controllers/user/alert/deployer/deployer.go

```diff
@@ -140,20 +141,28 @@ func (d *Deployer) sync() error {
 			if err != nil {
 				return fmt.Errorf("update cluster %v failed, %v", d.clusterName, err)
 			}
+
+			cluster, err := d.clusters.Get(d.clusterName, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("get cluster %s failed, %v", d.clusterName, err)
+			}
+			newCluster = cluster.DeepCopy()
 		}
 
 		if d.alertManager.IsDeploy, err = d.appDeployer.deploy(appName, appTargetNamespace, systemProjectID, systemProjectCreator); err != nil {
 			return fmt.Errorf("deploy alertmanager failed, %v", err)
 		}
 
-		return d.appDeployer.isDeploySuccess(newCluster, alertutil.GetAlertManagerDaemonsetName(appName), appTargetNamespace)
-	}
-
-	if d.alertManager.IsDeploy, err = d.appDeployer.cleanup(appName, appTargetNamespace, systemProjectID); err != nil {
-		return fmt.Errorf("clean up alertmanager failed, %v", err)
-	}
-	if mgmtv3.ClusterConditionAlertingEnabled.IsTrue(newCluster) {
-		mgmtv3.ClusterConditionAlertingEnabled.False(newCluster)
+		if err = d.appDeployer.isDeploySuccess(newCluster, alertutil.GetAlertManagerDaemonsetName(appName), appTargetNamespace); err != nil {
+			return err
+		}
+	} else {
+		if d.alertManager.IsDeploy, err = d.appDeployer.cleanup(appName, appTargetNamespace, systemProjectID); err != nil {
+			return fmt.Errorf("clean up alertmanager failed, %v", err)
+		}
+		if mgmtv3.ClusterConditionAlertingEnabled.IsTrue(newCluster) {
+			mgmtv3.ClusterConditionAlertingEnabled.False(newCluster)
```

Comment:
I think we can always set `ClusterConditionAlertingEnabled` to false here.

Reply:
updated